### DISCUSSION TITLE
Is it possible to inject one application service into another application service

### DISCUSSION THREAD

**Discussion by rajasingh012** [abpframework/abp#15343] [Q&A] [answered]
If I have code like this

```
public class ProjectAppService : CrudAppService<Project, ProjectDto, Guid, PagedAndSortedResultRequestDto, ProjectDto>, IProjectAppService 
{
}

public class IssueAppService : CrudAppService<Issue, IssueDto, Guid, PagedAndSortedResultRequestDto, IssueDto>, IIssueAppService 
{
}

```

Is it possible to inject IssueAppService in ProjectAppService. I want to override GetAsync in ProjectAppService so the return JSON should like

```
{
    "Name" : "New project",
   "Description" : " wonderful project Description"
  " Issues" : [ 
                  { "Name" : "Login not working", "Description" : " If we follow these steps, login screen not loading properly" },
                  { "Name" : "Signout not working", "Description" : " If we follow these steps, signout screen not loading properly" },
    ]
}
```

**Comment by EngincanV** [comment, marked as answer]
Yes, it's possible but I would not suggest it, instead use the domain layer for the required task (for example repositories). 

Let me share a quote from [Application Services Best Practices & Conventions](https://docs.abp.io/en/abp/latest/Best-Practices/Application-Services#using-other-application-services) document:

> Do not use other application services of the same module/application. Instead;
> * Use domain layer to perform the required task.
> * Extract a new class and share between the application services to accomplish the code reuse when necessary. But be careful to don't couple two use cases. They may seem similar at the beginning, but may evolve to different directions by time. So, use code sharing carefully.

**Reply by rajasingh012** [reply]
Thanks, Do you have code samples. Are you recommend using [DomainServices](https://docs.abp.io/en/abp/latest/Domain-Services) for this scenario?

**Reply by EngincanV** [reply]
Actually, It depends on your business logic and how you configured your application. If you have a domain service and thought the logic can not be fulfilled if you do not use one of its methods, then I would say go with it.
